### PR TITLE
Feature/xonotic module

### PR DESCRIPTION
**Pull request by kloenk**
###### Motivation for this change
provide xonotic as a module to host servers. This module is designed to host more than one server

###### Things done
create a xonotic service

This PR is dependet on #83461 for the usage of the xonotic-dedicated package

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/games/xonotic.nix [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      };
+    };
+  });
+in {
+  options = {
+    services.xonotic = {
+      enable = mkEnableOption "Xonotic servers";
+
+      servers = mkOption {
+        type = types.loaOf serverModule;
```
Comment (petabyteboy) [inline nixos/modules/services/games/xonotic.nix]: loaOf is deprecated

File: nixos/modules/services/games/xonotic.nix [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        default = {};
+        description = "available serves";
```
Comment (petabyteboy) [inline nixos/modules/services/games/xonotic.nix]: typo?

File: nixos/modules/services/games/xonotic.nix [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      serviceConfig = {
+        ExecStart = "${pkgs.xonotic-dedicated}/bin/xonotic-dedicated -userdir ${userDir}";
+        Restart = "always";
+        User = "xonotic";
+        WorkingDirectory = userDir;
+      };
+
+      preStart = let
+          configFile = pkgs.writeText "server.cfg" ''
```
Comment (petabyteboy) [inline nixos/modules/services/games/xonotic.nix]: weird indentation

File: nixos/modules/services/games/xonotic.nix [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          sv_public 1
+        '';
+        default = "";
+      };
+    };
+  });
+in {
+  options = {
+    services.xonotic = {
+      enable = mkEnableOption "Xonotic servers";
```
Comment (petabyteboy) [inline nixos/modules/services/games/xonotic.nix]: Can't you just check if cfg.servers != {}?

File: nixos/modules/services/games/xonotic.nix [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    options = {
+  
+      hostName = mkOption {
+        type = types.str;
+        description = "name of the server";
+        default = name;
+      };
+
+      port = mkOption {
+        type = types.ints.u16;
```
Comment (petabyteboy) [inline nixos/modules/services/games/xonotic.nix]: There's types.port

File: nixos/modules/services/games/xonotic.nix [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        description = "server port";
+        default = 26000;
+      };
+
+      options = mkOption {
```
Comment (petabyteboy) [inline nixos/modules/services/games/xonotic.nix]: More like extraConfig?

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by Kloenk** [comment]
Created a new xonotic module. will maybe upstream it later.
https://github.com/Kloenk/nix/commit/add9f5ad2760fdb96ed9534a3f88eacc88413d63

**Pull request closed without merging**